tests/core/test_IOManager.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import print_function
+import mock
+import unittest
+import os
+import threading
+import select
+
+#Make test version- agnostic:
+from sys import version_info
+if version_info.major == 2:
+  import __builtin__ as builtins    # pylint:disable=import-error
+else:
+  import builtins    # pylint:disable=import-error
+
+import IOManager
+
+
+class PipeWrapper:
+  def __init__(self):
+    read_fd, write_fd = os.pipe()
+    print("pipe read_fd: " + str(read_fd))
+    self.writer = os.fdopen(write_fd, 'w')
+    self.reader = os.fdopen(read_fd, 'r')
+
+  def send(self, msg):
+    self.writer.write(msg)
+    self.writer.flush()
+
+
+class TestIOManager(unittest.TestCase):
+  def setUp(self):
+    self.manager = IOManager.IOManager()
```

Comment:
by importing the class rather than the module, this becomes
self.manager = IOManager()